python/lsst/ts/tcpip/json_reading_one_client_server.py

```diff
@@ -0,0 +1,71 @@
+# This file is part of ts_tcpip.
+#
+# Developed for the Rubin Observatory Telescope and Site System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import abc
+import json
+import typing
+
+from .constants import TERMINATOR
+from .message_reading_one_client_server import MessageReadingOneClientServer
+
+__all__ = ["JsonReadingOneClientServer"]
+
+
+class JsonReadingOneClientServer(MessageReadingOneClientServer):
+    async def handle_message(self, message: str) -> None:
+        """Handle a message.
+
+        The message was read by the reader of this server.
+
+        Parameters
+        ----------
+        message : `str`
+            The message that was read by the reader of this server.
+        """
+        json_dict = json.loads(message)
+        self.log.debug(f"Read {json_dict=}")
+        await self.handle_json_dict(json_dict=json_dict)
+
+    @abc.abstractmethod
+    async def handle_json_dict(self, json_dict: dict[str, typing.Any]) -> None:
+        """Handle a JSON dict.
+
+        The dict results from deserializing a JSON document, which was read by
+        the reader of this server.
+
+        Parameters
+        ----------
+        json_dict : `dict` of `any`
+            The dict resulting from deserializing a JSON string.
+        """
+        raise NotImplementedError()
+
+    async def write_json_dict(self, json_dict: dict[str, typing.Any]) -> None:
```

Comment:
I had second thoughts about this. I now lean towards moving this method to the base class (your new plain-old-text-messages version), perhaps "write_data_as_json" or "write_message_as_json". After all, the TMA reads a custom format and writes json. So we even have a concrete use case.

Also note that the code does not require a dict -- any kind of data that can be json-encoded will work fine. I'm not sure we have a use case for other data types, but I still lean towards making the argument `data: typing.Any` or `message: typing.Any` because you lose nothing. Your doc string could say that the data is typically a dict, but that any json-encodable type is acceptable.